### PR TITLE
add nebula-console package action

### PR DESCRIPTION
**Pull request by czpmango**
## What type of PR is this?
- [ ] bug
- [ ] feature
- [x] enhancement

#### Issue(s) number: 
https://github.com/vesoft-inc/nebula/issues/3173

#### Description:
Package nebula-console into an RPM /deb package through action.

## Checklist:
Tests:
- [ ] Unit test(positive and negative cases)
- [ ] Function test
- [ ] Performance test
- [ ] N/A

Affects:
- [x] Documentation affected (Please add the label if documentation needs to be modified.)
- [ ] Incompatibility (If it breaks the compatibility, please describe it and add the label.）
- [ ] If it's needed to cherry-pick (If cherry-pick to some branches is required, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

## Release notes:
 - The RPM/DEB package will contain the nebula-console binary

### CODE REVIEW

File: src/console/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    FILES
+        ${CMAKE_BINARY_DIR}/nebula-console
+    PERMISSIONS
+        OWNER_EXECUTE OWNER_WRITE OWNER_READ
+        GROUP_EXECUTE GROUP_READ
+        WORLD_EXECUTE WORLD_READ
+    DESTINATION
+        bin
+    COMPONENT
+        console
```
Comment (yixinglu) [inline src/console/CMakeLists.txt:26]: prefer `nebula-console`

Comment (czpmango) [reply src/console/CMakeLists.txt]: ACK.

File: CMakeLists.txt [hunk 2]
```diff
@@ -25,6 +25,7 @@ project("Nebula Graph" C CXX)
 
 option(ENABLE_PACK_ONE "Whether to package into one" ON)
 option(ENABLE_VERBOSE_BISON "Enable Bison to report state" OFF)
+option(ENABLE_CONSOLE_COMPILATION "Enable nebula-console comppilation" OFF)
```
Comment (yixinglu) [inline CMakeLists.txt:28]: typo: comppilation

File: src/console/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    COMMAND ${GIT_EXECUTABLE} symbolic-ref --short HEAD 
+    OUTPUT_VARIABLE git_branch
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+    ERROR_QUIET
+    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+  )
+
+  add_custom_command(
+    OUTPUT ${CMAKE_BINARY_DIR}/nebula-console
+    COMMAND bash ${CMAKE_SOURCE_DIR}/src/console/compile-console.sh ${git_branch} ${CMAKE_BINARY_DIR}
```
Comment (yixinglu) [inline src/console/CMakeLists.txt:12]: what happens if there not exists related branch in nebula-console repo?

File: CMakeLists.txt (hunk 2 above)
Comment (czpmango) [reply CMakeLists.txt]: Fixed.

File: src/console/CMakeLists.txt (hunk 3 above)
Comment (czpmango) [reply src/console/CMakeLists.txt:12]: Compilation failed.

Comment (czpmango) [reply src/console/CMakeLists.txt:12]: Controlled by the cmake option `ENABLE_CONSOLE_COMPILATION`.

**Review by jievince — APPROVED** [review]

**Review by yixinglu — APPROVED** [review]

**Pull request merged**